spec/spec/fx.js

```diff
@@ -120,54 +144,83 @@ describe('FX', function() {
 
   describe('speed()', function() {
     it('set the speed of the animation', function(){
-      fx.speed(2)
-      expect(fx.spd).toBe(2)
-      expect(fx.situation.finish-fx.situation.start).toBe(500/2)
+      var dur, spd
 
-      fx.speed(0.5)
-      expect(fx.spd).toBe(0.5)
-      expect(fx.situation.finish-fx.situation.start).toBe(500/0.5)
+      dur = fx.situation.duration
 
-      fx.at(0.2).speed(2)
-      expect(fx.spd).toBe(2)
-      expect(fx.situation.finish-fx.situation.start).toBe(500/2)
+      spd = 2
+      fx.speed(spd)
+      expect(fx.spd).toBe(spd)
+      expect(fx.situation.finish-fx.situation.start).toBe(dur/spd)
 
-      fx.speed(1)
-      expect(fx.spd).toBe(1)
-      expect(fx.situation.finish-fx.situation.start).toBe(500)
+      spd = 0.5
+      fx.speed(spd)
+      expect(fx.spd).toBe(spd)
+      expect(fx.situation.finish-fx.situation.start).toBe(dur/spd)
+
+      spd = 2
+      fx.at(0.2).speed(spd)
+      expect(fx.spd).toBe(spd)
+      expect(fx.situation.finish-fx.situation.start).toBe(dur/spd)
+
+      spd = 1
+      fx.speed(spd)
+      expect(fx.spd).toBe(spd)
+      expect(fx.situation.finish-fx.situation.start).toBe(dur)
     })
 
     it('should not change the position when the animation is run backward', function(){
-      expect(fx.at(0.4).reverse(true).speed(2).pos).toBe(0.4)
+      var pos = 0.4
+
+      expect(fx.at(pos).reverse(true).speed(2).pos).toBe(pos)
     })
 
     it('return the current speed with no argument given', function(){
-      fx.speed(2)
-      expect(fx.speed()).toBe(2)
+      var spd
+
+      spd = 2
+      fx.speed(spd)
+      expect(fx.speed()).toBe(spd)
+
+      spd = 0.5
+      fx.speed(spd)
+      expect(fx.speed()).toBe(spd)
 
-      fx.speed(0.5)
-      expect(fx.speed()).toBe(0.5)
+      spd = 1
+      fx.speed(spd)
+      expect(fx.speed()).toBe(spd)
+    })
 
-      fx.speed(0)
-      expect(fx.speed()).toBe(0.5)
+    it('pause the animation when a speed of 0 is passed', function(){
+      var spd = fx.speed()
 
-      fx.speed(1)
-      expect(fx.speed()).toBe(1)
+      expect(fx.speed(0)).toBe(fx)
+      expect(fx.speed()).toBe(spd)
+      expect(fx.paused).toBe(true)
     })
```

Comment:
You call 3 times speed in this test while testing for one specific call. That could lead to tests missbehaving.
I suggest just calling `speed(0)` and check if it is paused.

```js
var ret = fx.speed(0)
expect(ret).toBe(fx)
expect(fx.paused).toBe(True)
```